### PR TITLE
Add MyGet instructions to use dev build

### LINKED ISSUE #2025: Update instructions to use MyGet for dev build
## Feature Request

And optionally, deprecate `master` dist-tag.

### Package history

Package history at [MyGet](https://botbuilder.myget.org/feed/botframework-webchat/package/npm/botframework-webchat).

### `.npmrc`

```
registry=https://botbuilder.myget.org/F/botframework-webchat/npm/
```

### Rough draft

To use dev build,

1. Create/modify `.npmrc` locally and set `registry`
1. Run `npm install botframework-webchat --save`
1. Remove your changes in step 1 to reduce impact on your `package-lock.

@corinagum please verify these steps with a few things.

1. Look at `package-lcok.json` to make sure we have minimal impact
1. Remove `node_modules`, keep `package-lock.json`, then do `npm install` again
   - This mimic what the user would do on a fresh clone
   - Make sure `npm install` is grabbing the right version by drilling into `node_modules` folder seeing the `package.json` file inside
1. Same as step 2, but instead of `npm install`, run` npm ci`
   - Should be the same
1. Wirte up guides to reverse it back to production version
   1. (I didn't try the following steps)
   1. Run `npm install botframework-webchat --save`
   1. Make sure everything in `package-lock.json` is pointing to npmjs.org, and *not a single package* is point to myget.org
   1. Make sure it grab the 4.4.1 (production, not dev build)
      - Drill into `node_modules` folder to see
      - Don't forget to look at `botframework-webchat-core` and other packages
   1. Find the string "myget.org" in `package-lock.json`, make sure no package is affected if the user reverted, super important

### PR DESCRIPTION
**Pull request by corinagum**
Fixes #2025 

 ## Changelog Entry	
- Added documentation for using Web Chat dev build, by [@corinagum](https://github.com/corinagum), in PR [#2074](https://github.com/Microsoft/BotFramework-WebChat/pull/2074)

 ## Description	
Our root README.md now has information on how to access Web Chat's dailies for the latest changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -219,15 +219,43 @@ There are several properties that you might pass into your Web Chat React Compon
...
 
-## Contributing
+# How to test with Web Chat's latest bits
+
+*Testing unreleased features is only available via MyGet packaging at this time.*
+
+If you want to test a feature or bug fix that has not yet been released, you will want to point your Web Chat package to Web Chat's daily feed, as opposed the official npmjs feed.
+
+Currently, you may access Web Chat's dailies by subscribing to our MyGet feed. To do this, you will need to update the registry in your project. **This change is reversable, and our directions include how to revert back to subscribing to the official release**.
```
Comment (compulim) [inline README.md]: Is it "reversible" or "reversable"?

```suggestion
Currently, you may access Web Chat's dailies by subscribing to our MyGet feed. To do this, you will need to update the registry in your project. **This change is reversible, and our directions include how to revert back to subscribing to the official release**.
```

File: README.md [hunk 2]
```diff
@@ -219,15 +219,43 @@ There are several properties that you might pass into your Web Chat React Compon
...
+
+To do this you may add your packages and then change the registry of your project.
+
+1. Add your project dependencies other than Web Chat.
+1. In your project's root directory, create a `.npmrc` file
+1. Add the following line to your file: `registry=https://botbuilder.myget.org/F/botframework-webchat/npm/`
+1. Add Web Chat to your project dependencies `npm i botframework-webchat --save`
+1. Note that in your `package-lock.json`, the registries pointed to are now MyGet. The Web Chat project has upstream source proxy enabled, which will redirect non-MyGet packages to `npmjs.com`.
+
+##  Re-subscribe to official release on `npmjs.com`
```
Comment (compulim) [inline README.md]:
```suggestion
## Re-subscribe to official release on `npmjs.com`
```

File: README.md [hunk 3]
```diff
@@ -219,15 +219,43 @@ There are several properties that you might pass into your Web Chat React Compon
...
+Re-subscribing requires that you reset your registry.
+
+1. Delete your `.npmrc file`
+1. Delete your root `package-lock.json`
```
Comment (compulim) [inline README.md]: Can we do it without removing `package-lock.json`?

Although data/information in `package-lock.json` is not quite human-readable, they are critical to make a reproducible build.

File: README.md [hunk 4]
```diff
@@ -219,15 +219,43 @@ There are several properties that you might pass into your Web Chat React Compon
...
+1. Remove your `node_modules` directory
+1. Clear your project cache `npm cache clean --force`
```
Comment (compulim) [inline README.md]: Just curious, do we really need to clean cache?

File: README.md [hunk 5]
```diff
@@ -219,15 +219,43 @@ There are several properties that you might pass into your Web Chat React Compon
...
+1. Reinstall your packages with `npm i`
+1. Note that in your `package-lock.json`, the registries are pointing to `npmjs.com` again.
```
Comment (compulim) [inline README.md]:
```suggestion
1. Note that in your `package-lock.json`, the registries are pointing to https://npmjs.com/ again.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,6 +26,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 -  `*`: Added [`eslint`](https://npmjs.com/package/eslint/) to static code analysis, by [@compulim](https://github.com/compulim), in PR [#1970](https://github.com/microsoft/BotFramework-WebChat/pull/1970)
 -  Added pt-PT language, by [@bodyzatva](https://github.com/bodyzatva) in PR [#2005](https://github.com/microsoft/BotFramework-WebChat/pull/2005) and PR [#2020](https://github.com/Microsoft/BotFramework-WebChat/pull/2020)
+- Added documentation for using Web Chat dev build, by [@corinagum](https://github.com/corinagum), in PR [#2074](https://github.com/Microsoft/BotFramework-WebChat/pull/2074)
```
Comment (compulim) [inline CHANGELOG.md]: Not now, but later, should we section out non-code change to a separate section in the CHANGELOG.md?

File: README.md (hunk 3 above)
Comment (corinagum) [reply README.md]: Tried without deleting `package-lock.json` and it did not work.

File: README.md (hunk 4 above)
Comment (corinagum) [reply README.md]: Looks like it works without. :) thanks

**Comment by tiagodenoronha** [comment]
@corinagum was this related to #2091 ? :)

**Comment by corinagum** [comment]
@tiagodenoronha I'm not sure -- I believe you filed 2091 after this PR was merged in. Please feel free to review the documentation though! Thank you.

**Pull request merged**